mava/utils/training_utils.py

```diff
@@ -6,6 +6,106 @@
 import tensorflow as tf
 import trfl
 
+from mava.types import NestedArray
+
+
+# Adapted from
+# https://github.com/tensorflow/agents/blob/f2bebb1e3bc34dc49e34a3d1d6baf30ceee9523c/tf_agents/utils/value_ops.py#L98
+def generalized_advantage_estimation(
```

Comment:
What are the main adaptations here? I.e. why are we not using the originals?

Reply:
I think we were calculating the `empirical_advantage` and not the gae advantage before and `tf.scan` and is faster than what were doing before.